spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationJob.scala

```diff
@@ -298,11 +298,14 @@ object StandardizationJob {
                                dao: MenasDAO): DataFrame = {
     val numberOfColumns = schema.fields.length
     val dfReaderConfigured = getFormatSpecificReader(cmd, dataset, numberOfColumns)
-    val dfWithSchema = (if (!cmd.rawFormat.equalsIgnoreCase("parquet")
+    val dfWithSchema = (if (cmd.rawFormat.equalsIgnoreCase("fixed-width")) {
```

Comment:
I don't see the point and find it uglier than plain simple ifs. But I don't mind switching it up.

Do you find this more readable:

```scala
    val finalReader = cmd.rawFormat.toLowerCase() match {
      case "fixed-width" =>
        val inputSchema = PlainSchemaGenerator.generateInputSchema(schema)
        dfReaderConfigured.schema(inputSchema)
      case format if (format != "parquet" && format != "cobol") =>
        // SparkUtils.setUniqueColumnNameOfCorruptRecord is called even if result is not used to avoid conflict
        val columnNameOfCorruptRecord = SparkUtils.setUniqueColumnNameOfCorruptRecord(spark, schema)
        val optColumnNameOfCorruptRecord: Option[String] = if (cmd.failOnInputNotPerSchema) {
          None
        } else {
          Option(columnNameOfCorruptRecord)
        }
        val inputSchema = PlainSchemaGenerator.generateInputSchema(schema, optColumnNameOfCorruptRecord)
        dfReaderConfigured.schema(inputSchema)
      case _ => dfReaderConfigured
    }

```

Reply:
What about:
```
    val finalReader = cmd.rawFormat.toLowerCase() match {
      case "fixed-width" =>
        val inputSchema = PlainSchemaGenerator.generateInputSchema(schema)
        dfReaderConfigured.schema(inputSchema)
      case "parquet" || "cobol" => dfReaderConfigured
      case _ =>
        // SparkUtils.setUniqueColumnNameOfCorruptRecord is called even if result is not used to avoid conflict
        val columnNameOfCorruptRecord = SparkUtils.setUniqueColumnNameOfCorruptRecord(spark, schema)
        val optColumnNameOfCorruptRecord: Option[String] = if (cmd.failOnInputNotPerSchema) {
          None
        } else {
          Option(columnNameOfCorruptRecord)
        }
        val inputSchema = PlainSchemaGenerator.generateInputSchema(schema, optColumnNameOfCorruptRecord)
        dfReaderConfigured.schema(inputSchema)
```

And many-line default case could be a function, if really to make it compact 😉

Reply:
I tried something, rewrote it even more. Please have a look